components/wifi-controller/CMakeLists.txt

```diff
@@ -1,8 +1,8 @@
 idf_component_register(
     SRCS 
-        "wifi-controller.c"
+        "espidf-v4.1.c"
```

Comment:
created two versions of wifi-controller.c: one using API version 4.0 and one using API version 4.1.
4.1 is identical to wifi-controller.c and used for "latest" esp-idf.
4.0 is used for PlatformIO, as PlatformIOuses esp-idf 4.0.1